draft-ietf-tls-tls13.md

```diff
@@ -2969,24 +2968,33 @@ MUST generate a fatal "unexpected_message" alert.  After sending a
 KeyUpdate message, the sender SHALL send all its traffic using the
 next generation of keys, computed as described in
 {{updating-traffic-keys}}. Upon receiving a KeyUpdate, the receiver
-MUST update their receiving keys and if they have not already updated
-their sending state up to or past the then current receiving
-generation MUST send their own KeyUpdate prior to sending any other
-messages.  This mechanism allows either side to force an update to the
-entire connection. Note that implementations may receive an arbitrary
+MUST update its receiving keys.
+
+Each implementation MUST receive a min_send_generation counter that
+is initialized to 0 at the time the traffic keys are installed.
+Upon receiving a KeyUpdate, if the receiver has sent any data records since receiving the
+last KeyUpdate it MUST also increment its min_send_generation value
+by 1. Otherwise, the min_send_generation MUST remain unchanged. Prior
+to sending a record, if min_send_generation is greater than the
+current sending generation, the sender MUST first send a KeyUpdate.
```

Comment:
I believe the intended interpretation was that min_send_generation *only* gets bumped on incoming KeyUpdate and not when you initiate KeyUpdate, so it's not quite a flag. If you initiate KeyUpdate, then min_send = send - 1 and the peer's KeyUpdate will catch you up.

The failure mode is then the inverse because sending 1,000 KeyUpdates for fun will get min_send too far behind send to catch up easily, especially with coallescing.